src/test/java/uk/gov/dhsc/htbhf/utils/NinoGenerator.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.dhsc.htbhf.utils;
+
+import com.google.common.base.Strings;
+
+/**
+ * Utility methods to generate random NINOs.
+ */
+public class NinoGenerator {
+
+    /**
+     * Represents 1 child under 1 and 2 under 4 to make the value returned from the stub consistent
+     */
+    private static final String CHILDREN_NUMBER = "12";
+    private static final String NINO_PREFIX_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    private static final String ELIGIBLE_CHAR = "E";
+    private static final String NINO_SUFFIX_CHARS = "ABC";
+
+    public static String generateEligibleNino() {
+        return randomEligibleTwoChars() + CHILDREN_NUMBER + randomFourDigitInteger() + randomCharAtoC();
+    }
+
+    private static String randomFourDigitInteger() {
+        int number = (int) Math.floor((Math.random() * (9999)) + 1);
+        return Strings.padStart(String.valueOf(number), 4, '0');
+    }
+
+    private static char randomCharFromChars(String allChars) {
+        double randomCharPosition = Math.floor(Math.random() * allChars.length());
```

Comment:
Can I suggest using a `new Random()` as an instance field instead of `Math.random()`, then you can use `random.nextInt(allChars.length())`.

Reply:
Yes, this is much better now, should have spotted that, thanks.